.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,39 @@
+name: CodeQL Analysis
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 8 * * *'
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: windows-latest
+    steps:
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.0.2
+
+      - name: Checkout repository
+        id: checkout_repo
+        uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        id: init_codeql
+        uses: github/codeql-action/init@v1
+        with:
+          language: csharp
+          queries: security-and-quality
+
+      - name: Build solution
+        id: build_solution
+        run: |
+          nuget restore .\TestFx.sln
+          msbuild .\TestFx.sln /p:UseSharedCompilation=false
```

Comment:
```suggestion
          .\build.cmd -configuration Debug -Full
```

Reply:
Without the property `/p:UseSharedCompilation=false` applied to the build process, CodeQL doesn't recognize the build and [throws an error](https://github.com/aleks-ivanov/testfx/runs/2819883514?check_suite_focus=true).

If the script is mandatory, a possible solution might be a custom parameter that accepts values and applies to the build command.

If the script is not mandatory, we think it would be easier to implement and maintain, just to use the `nuget restore` and `msbuild` commands with the required properties like configuration, platform etc. so the `UseSharedCompilation` property could also be added, something like this:

`msbuild .\TestFx.sln /p:Configuration=Debug /p:Platform='Any CPU' /p:UseSharedCompilation=false`

Reply:
Can you enable commits from maintainers to these PRs? I can fix the scripts to support this.

Reply:
@Haplois since I've created this PR from our organization, this option is ignored by default. To speed up the process we've made you a maintainer to the fork, so after you accept the invitation, you should be able to commit to this PR 🙂

Reply:
@aleks-ivanov I think we might still need to rely on `build.cmd` because simply doing nuget restore doesn't restore enough of the dependencies we need.